packages/rrweb-snapshot/src/rebuild.ts

```diff
@@ -78,6 +80,79 @@ export function createCache(): BuildCache {
   };
 }
 
+/**
+ * undo splitCssText/markCssSplits
+ * (would move to utils.ts but uses `adaptCssForReplay`)
+ */
+export function applyCssSplits(
+  n: serializedElementNodeWithId,
+  cssText: string,
+  hackCss: boolean,
+  cache: BuildCache,
+): void {
+  const childTextNodes: serializedTextNodeWithId[] = [];
+  for (const scn of n.childNodes) {
+    if (scn.type === NodeType.Text) {
+      childTextNodes.push(scn);
+    }
+  }
+  for (let i = 0; i < childTextNodes.length; i++) {
+    let remainder = '';
+    const scn = childTextNodes[i];
+    if (i !== childTextNodes.length - 1) {
+      const ix = cssText.indexOf('/* rr_split */');
+      if (ix !== -1) {
+        remainder = cssText.substring(ix + '/* rr_split */'.length);
+        cssText = cssText.substring(0, ix);
+      }
+    } else {
+      // todo: replaceAll after lib.es2021
+      cssText = cssText.replace(/\/\* rr_split \*\//g, '');
+    }
+    if (hackCss) {
+      cssText = adaptCssForReplay(cssText, cache);
+    }
+    // id will be assigned when these child nodes are
+    // iterated over in buildNodeWithSN
+    scn.textContent = cssText;
+    cssText = remainder;
+  }
```

Comment:
This seems like it would be a lot simpler by splitting the cssText on `/* rr_split */`:
```suggestion
  const cssTextSplits = cssText.split("/* rr_split */")
  for (let i = 0; i < childTextNodes.length; i++) {
    const childTextNode = childTextNodes[i];
    const cssTextSection = cssTextSplits[i];
    if (childTextNode && cssTextSection) {
      // id will be assigned when these child nodes are
      // iterated over in buildNodeWithSN
      childTextNode.textContent = hackCSS ? adaptCssForReplay(cssTextSection, cache) : cssTextSection;
    }
  }
```

Reply:
The refined code is easier to understand to me

Reply:
@eoghanmurray it would be great if we could add this code before merging